File: backend/src/models/Policy.js
const mongoose = require("mongoose");

const policySchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: { type: String, required: true },
  targetRegions: [{ type: String }],
  policyCode: { type: String, required: true, unique: true },
  startDate: { type: Date, required: true },
  endDate: { type: Date, required: true },
  status: {
    type: String,
    enum: ["draft", "published", "active", "paused", "closed"],
    default: "draft",
  },
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User",
    required: true,
  },
  createdAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model("Policy", policySchema);

File: mobile/civic_engagement_app/lib/src/features/citizen/domain/entities/policy.dart
import 'package:equatable/equatable.dart';

class Policy extends Equatable {
  const Policy({
    required this.id,
    required this.title,
    required this.description,
    required this.policyCode,
    required this.targetRegions,
    required this.status,
    required this.averageRating,
    required this.totalVotes,
    this.startDate,
    this.endDate,
    this.createdBy,
    this.createdAt,
  });

  final String id;
  final String title;
  final String description;
  final String policyCode;
  final List<String> targetRegions;
  final DateTime? startDate;
  final DateTime? endDate;
  final String status;
  final double averageRating;
  final int totalVotes;
  final String? createdBy;
  final DateTime? createdAt;

  bool get canVote => status == 'active';
  bool get isPaused => status == 'paused';

  @override
  List<Object?> get props => [
    id,
    title,
    description,
    policyCode,
    targetRegions,
    startDate,
    endDate,
    status,
    averageRating,
    totalVotes,
    createdBy,
    createdAt,
  ];
}
